control/tests/margin_test.py

```diff
@@ -11,6 +11,22 @@
 from control.statesp import StateSpace
 from control.margins import *
 
+def assert_array_almost_equal(x, y, ndigit=4):
```

Comment:
A comment indicating why you need this would be good.  I thought perhaps it was because nan's compare unequal, but np.testing.assert_array_almost_equal seems to handle nans fine (i.e., considers them to be equal).